lib/pluginmanager/util.rb

```diff
@@ -37,6 +37,36 @@ def self.logstash_plugin?(plugin, version = nil)
     end
   end
 
+  # Fetch latest version information as in rubygems
+  # @param [String] The plugin name
+  # @param [Hash] Set of available options when fetching the information
+  # @option options [Boolean] :pre Include pre release versions in the search (default: false)
+  # @return [Hash] The plugin version information as returned by rubygems
+  def self.fetch_latest_version_info(plugin, options={})
+    require "gems"
+    options = { :pre => false }.merge(options)
+    versions = Gems.versions(plugin)
+    if !options[:pre]
+      versions.select! { |version| !version["prerelease"] }
+    end
+    versions[0]
+  end
+
+  # Let's you decide to update to the last version of a plugin if this is a major version
+  # @param [String] A plugin name
+  # @return [Boolean] True in case the update is moving forward, false otherwise
+  def self.update_to_major_version?(plugin_name)
+    plugin_version  = fetch_latest_version_info(plugin_name)
+    latest_version  = plugin_version['number'].split(".")
+    current_version = Gem::Specification.find_by_name(plugin_name).version.version.split(".")
+    if (latest_version[0].to_i > current_version[0].to_i)
+      ## warn if users want to continue
+      puts("You are updating #{plugin_name} to a new version #{latest_version.join('.')}, which may not be compatible with #{current_version.join('.')}. are you sure you want to proceed (Y/N)?")
+      return ( "y" == STDIN.gets.strip.downcase ? true : false)
+    end
+    true
+  end
```

Comment:
That might make sense, however what is still not clear to me is that this class you speak about is under logstash, but when we're in the plugin manager we're not supposed to load stuff under there. At less this is what I understood while working on #3404 (about the pack/unpack commands you can see in the utils module there, located under bootstrap).

I agree we should have this kind of things unified, this is a great idea for a refactoring where we can handle this kind of code only in one place.

What do you think?